common/lib/types/message.js

```diff
@@ -172,17 +186,35 @@ var Message = (function() {
 								throw new Error('Unable to decrypt message; not an encrypted channel');
 							}
 						default:
+							var codec = context.codecs[xform];
+							if(codec) {
+								try {
+									data = codec.decode(data, {
+										channelOptions: context.channelOptions,
+										encoding: match[0],
+										baseEncodedPreviousPayload: context.baseEncodedPreviousPayload
+									});
+									if(xform === 'vcdiff') {
```

Comment:
Why is this only done for `vcdiff`?  If we introduced another codec, why would we not store the last payload?

Reply:
This is `vcdiff` specific sadly. We operate under the assumption that we need to store the payload right after we decoded `vcdiff` encoding if it was there and before we removed any other encoding. We also operate under the assumption that there might be only `base64` after `vcdiff`